storage/FlashStorage.h

```diff
@@ -36,13 +36,18 @@
 #ifndef UBIRCH_FLASH_STORAGE_H
 #define UBIRCH_FLASH_STORAGE_H
 
+#include <cstdio>
+
 #ifdef __cplusplus
```

Comment:
extern "C" {} is for linking C into C++, this will do nothing, except you want to include fstorage which is a C library, so just put it around the fstorage.h include if its not done inside.

Reply:
done